sparkly/utils.py

```diff
@@ -59,6 +64,89 @@ def absolute_path(file_path, *rel_path):
     )
 
 
+def chunks(seq, n):
+    """Iterate over seq by chunks of n elements.
+
+    >>> list(chunks([1, 2, 3, 4, 5], 2))
+    [[1, 2], [3, 4], [5]]
+    >>> list(chunks([], 2))
+    []
+    >>> list(chunks([1], 2))
+    [[1]]
+    """
+    it = iter(seq)
+    while True:
+        chunk = list(islice(it, n))
+        if not chunk:
+            break
+        yield chunk
+
+
+def data_type_defines(t, required_fields):
```

Comment:
I prefer to keep `dataType` in there since it's really working with `dataType`s and not arbitrary python types, but I'll change it to `_has`.